src/app/features/courses/courses.component.ts

```diff
@@ -1,8 +1,47 @@
-import { Component } from "@angular/core";
+import { Component, OnInit } from "@angular/core";
+import { Router } from "@angular/router";
+import { CoursesStoreService } from "@app/services/courses-store.service";
+
+import { Course } from "@app/shared/models/course.model";
+import { UserStoreService } from "@app/user/services/user-store.service";
 
 @Component({
   selector: "app-courses",
   templateUrl: "./courses.component.html",
   styleUrls: ["./courses.component.scss"],
 })
-export class CoursesComponent {}
+export class CoursesComponent implements OnInit {
+  constructor(
+    private router: Router,
+    private coursesStoreService: CoursesStoreService,
+    private userStorageService: UserStoreService
+  ) {}
+  // info page text
+  infoTitle = "Your List Is Empty";
+  infoText = "Please use 'Add New Course' button to add your first course";
+
+  courses: Course[] = [];
+  loading: boolean = false;
```

Comment:
This is currently not used, I guess it will be once other functionality is implemented.